### PR TITLE
Fixed probable bugs (Fixed #1317)

### LINKED ISSUE #1317: Fix probable bugs
[Parent issue #1311 ]

**Actual Behaviour**

Currently, there are a number of warnings on running lint check that might produce exceptions :
- Constant conditions and expressions that are always true or false
- Problems related to @Nullable and @NotNull annotations usage in constant conditions & exceptions
- Unused assignments

**Expected Behaviour**

These conditions might result in exceptions like Null Pointer Exception. Hence, there is a need to use proper annotations and a need to handle conditions and expressions properly to avoid probable bugs.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by arundhati24**
Fixes #1317 [Parent #1311 ]

Changes: Added appropriate annotations and checked for nullability conditions, wherever required.

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java [hunk 1]
```diff
@@ -221,6 +222,7 @@ public void onDetachedFromRecyclerView(RecyclerView recyclerView) {
     public int getItemViewType(int position) {
         ChatMessage item = getItem(position);
 
+        assert item != null;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: Never use assert in production code

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: We don't need to throw exceptions. We need to check if they are null and handle it gracefully

If they should not be null and they are, then well any other exception is not much better than NullPointerException

Mostly, exceptions are good in service layer, never in view layer

**Comment by arundhati24** [comment]
@iamareebjamal Made the changes.

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ChatViewHolder.java [hunk 2]
```diff
@@ -244,12 +258,16 @@ public void onFailure(Call<SkillRatingResponse> call, Throwable t) {
...
-                        setRating(false, true);
+                        if(thumbsUp != null) {
+                            thumbsUp.setImageResource(R.drawable.thumbs_up_outline);
+                            setRating(false, true);
+                        }
                         break;
                     case Constant.NEGATIVE:
-                        thumbsDown.setImageResource(R.drawable.thumbs_down_outline);
-                        setRating(false, false);
+                        if(thumbsDown != null) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ChatViewHolder.java]: Space after if is missing everywhere

File: app/src/main/java/ai/kitt/snowboy/AppResCopy.java [hunk 3]
```diff
@@ -45,7 +45,7 @@ private static void copyFilesFromAssets(Context context, String assetsSrcDir, St
                 InputStream is = context.getAssets().open(assetsSrcDir);
                 FileOutputStream fos = new FileOutputStream(outFile);
                 byte[] buffer = new byte[1024];
-                int byteCount = 0;
+                int byteCount ;
```
Comment (iamareebjamal) [inline app/src/main/java/ai/kitt/snowboy/AppResCopy.java]: Space after ; should not be there

**Pull request merged**